File: client/src/pages/pegawai/types/servis.ts
export interface Barang {
  id_barang: number
  nama_barang: string
  harga: number
}

export interface DetailServisItem {
  id_detail?: number
  id_barang: number | null
  deskripsi: string
  jumlah: number
  harga_satuan: number
  biaya: number
}

export interface ServisData {
  id_servis?: number
  nama_pelanggan: string
  no_whatsapp: string
  tipe_hp: string
  keluhan: string
  status_servis: string
  tanggal_masuk: string
  tanggal_selesai: string | null
  biaya_servis: number
  biaya_total: number
  id_user: number
  detail: DetailServisItem[]
}

File: api/models/servis.go
package models

type Servis struct {
    IDServis       int             `json:"id_servis"`
    NamaPelanggan  string          `json:"nama_pelanggan"`
    NoWhatsapp     string          `json:"no_whatsapp"`
    TipeHP         string          `json:"tipe_hp"`
    Keluhan        string          `json:"keluhan"`
    StatusServis   string          `json:"status_servis"`
    BiayaServis    float64         `json:"biaya_servis"`
    BiayaTotal     float64         `json:"biaya_total"`
    TanggalMasuk   string          `json:"tanggal_masuk"`
    TanggalSelesai *string         `json:"tanggal_selesai"`
    Detail         []DetailServis  `json:"detail"`
}
